File: Data Structures & Algorithms/is-palindrome/submission-2.js
class Solution {

    isAlphaNum(ch) {

        if (
            (ch >= '0' && ch <= '9') ||
            (ch.toLowerCase() >= 'a' &&
             ch.toLowerCase() <= 'z')
        ) {
            return true;
        }

        return false;
    }

    isPalindrome(s) {

        let st = 0;
        let end = s.length - 1;

        while (st <= end) {

            if (!this.isAlphaNum(s[st])) {
                st++;
                continue;
            }

            if (!this.isAlphaNum(s[end])) {
                end--;
                continue;
            }

            if (
                s[st].toLowerCase() !==
                s[end].toLowerCase()
            ) {
                return false;
            }

            st++;
            end--;
        }

        return true;
    }
}

File: Data Structures & Algorithms/palindrome-number/submission-2.py
class Solution:

    def reverse(self, n: int) -> int:

        reverseNum = 0

        sign = -1 if n < 0 else 1

        n = abs(n)

        while n != 0:

            dig = n % 10

            # overflow check
            if reverseNum > (2**31 - 1) // 10:
                return 0

            reverseNum = reverseNum * 10 + dig

            n //= 10

        return reverseNum * sign


    def isPalindrome(self, n: int) -> bool:

        if n < 0:
            return False

        reverseNumber = self.reverse(n)

        return reverseNumber == n